includes/Modules/Sign_In_With_Google/Web_Tag.php

```diff
@@ -0,0 +1,215 @@
+<?php
+/**
+ * Class Google\Site_Kit\Modules\Sign_In_With_Google\Web_Tag
+ *
+ * @package   Google\Site_Kit\Modules\Sign_In_With_Google
+ * @copyright 2024 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Modules\Sign_In_With_Google;
+
+use Google\Site_Kit\Core\Modules\Tags\Module_Web_Tag;
+use Google\Site_Kit\Core\Util\BC_Functions;
+use Google\Site_Kit\Core\Util\Method_Proxy_Trait;
+use Google\Site_Kit\Modules\Sign_In_With_Google\Authenticator;
+
+/**
+ * Class for Web tag.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Web_Tag extends Module_Web_Tag {
+
+	use Method_Proxy_Trait;
+
+	/**
+	 * Module settings.
+	 *
+	 * @since n.e.x.t
+	 * @var Settings
+	 */
+	private $settings;
+
+	/**
+	 * Context instance.
+	 *
+	 * @since n.e.x.t
+	 * @var \Google\Site_Kit\Context
+	 */
+	private $context;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string                   $tag_id      Tag ID.
+	 * @param string                   $module_slug Module slug.
+	 * @param Settings                 $settings    Module settings instance.
+	 * @param \Google\Site_Kit\Context $context     Context instance.
+	 */
+	public function __construct( $tag_id, $module_slug, Settings $settings, \Google\Site_Kit\Context $context ) {
+		parent::__construct( $tag_id, $module_slug );
+		$this->settings = $settings;
+		$this->context  = $context;
```

Comment:
Most/all other module tags don't override the constructor like this but use setters to receive additional data from outside.